providers/default.rb

```diff
@@ -201,26 +201,42 @@ def windows?
 # gem files.
 #
 def move_opt_chef(src, dest)
-  converge_by("move all files under #{src} to #{dest}") do
-    FileUtils.rm_rf dest
-    raise "rm_rf of #{dest} failed" if ::File.exist?(dest) # detect mountpoints that were not deleted
-    FileUtils.mv src, dest
-  end
-rescue => e
-  # this handles mountpoints
-  converge_by("caught #{e}, falling back to copying and removing from #{src} to #{dest}") do
+  if windows?
```

Comment:
We probably want to platform?("windows") here so it works on older chef releases

Reply:
i'm pretty sure `windows?` is ancient Chef-10-era API from `lib/chef/platform/query_helpers.rb`

Reply:
`windows` is a def in this provider. That def is using `plaform_family`.
The syntax `windows?` is being used 7 other times in this very same library.